docs/application/native/guides/machine-learning/machine-learning-inference.md

```diff
@@ -298,6 +298,132 @@ Followings are the available elements:
+            input tensors                         output tensors
+            ```
+
+      - then-option: Option for TRUE action
+        * nth tensor, nth tensor, ... (for TENSORPICK option)
+
+      - else: Action if the result is FALSE
+        * PASSTHROUGH: The element will not make changes to the buffers, buffers are pushed straight through.
+        * SKIP: Do not generate output frame (frame skip)
+        * TENSORPICK: Choose nth tensor (or tensors) among input tensors
+
+      - else_option: Option for FALSE action
+        * nth tensor,nth tensor,... (for TENSORPICK option)
+
+    - **Example launch line with simple if-condition**
+
+    If a specific element ([3][4][2][5] value of the first tensor) is between a given range ([10,100]) in a tensor frame), it simply passes through the input tensors and if not, pushes the selected tensors onto the src-pad. It can be expressed as:
+
+    ```
+    gst-launch ... (tensors 0,1,2 stream) !
+        tensor_if name=tif \
+                    compared-value=A_VALUE compared-value-option=3:4:2:5,0 \
+                    supplied-values=10,100 \
+                    operator=RANGE_INCLUSIVE \
+                    then=PASSTHROUGH \
+                    else=TENSORPICK \
+                    else-option=0,2 \
+        ! tif.src_0 ! (tensors 0,1,2 for TRUE action) ...
+        ! tif.src_1 ! (tensors 0,2 for FALSE action) ...
+    ```
+
+    - **Custom if-condition**
+
+      If the if-condition is complex and cannot be expressed with tensor_if expressions, you can define custom condition.
+
+      Then, you can create a pipeline as follows:
+
+      - **Define and register if_custom**
+
+        Before you use the custom if-condition in the pipeline, you have to register the Custom if-condition with input tensors information and data:
+
+        ```c
+        /* Define custom if-condition */
+        static int tensor_if_custom_cb (const ml_tensors_data_h data, const ml_tensors_info_h info, int *result, void *user_data)
+        {
+        /* Describe the conditions and pass the result. Result 0 refers to FALSE and a non-zero value refers to TRUE. */
+        *result = 1;
+        /* Return 0 if there is no error. */
+        return 0;
+        }
+        ```
+
+      - **Construct a pipeline with custom if-condition**
+
+        After registering the custom if-condition, you can use it when constructing the pipeline:
+
+        ```c
+        /* The pipeline description (input data with dimension 2:1:1:1 and type int8 will be passed to tensor_if custom condition. Depending on the result, proceed to true or false paths.) */
+        const char pipeline[] = "appsrc ! other/tensor,dimension=(string)2:1:1:1,type=(string)int8,framerate=(fraction)0/1 ! tensor_if name=tif compared-value=CUSTOM compared-value-option=tif_custom_cb_name then=PASSTHROUGH else=PASSTHROUGH tif.src_0 ! tensor_sink name=true_condition async=false tif.src_1 ! tensor_sink name=false_condition async=false"
```

Comment:
then=PASSTHROUGH ?? Isn't something missing here?

Reply:
`then` does not require additional `then-option`.
The `tif.src_0` or `tif.src_1` path is determined by the condition result.